pihole

```diff
@@ -312,42 +312,55 @@ analyze_ports() {
 }
 
 statusFunc() {
-  # Determine if there is a pihole service is listening on port 53
-  local listening
-  listening="$(lsof -Pni:53)"
-  if grep -q "pihole" <<< "${listening}"; then
-    if [[ "${1}" != "web" ]]; then
-      analyze_ports "${listening}"
-    fi
+  # Determine if there is pihole-FTL service is listening on any UDP port
+  local listening pid port
+
+  pid="$(getFTLPID)"
+  if [[ "$pid" -eq "-1" ]]; then
+      case "${1}" in
+        "web") echo "-1";;
+        *) echo -e "  ${CROSS} DNS service is NOT running";;
+      esac
+      return 0
   else
-    case "${1}" in
-      "web") echo "-1";;
-      *) echo -e "  ${CROSS} DNS service is NOT listening";;
-    esac
-    return 0
-  fi
+      #get the port pihole-FTL is listening on
+      port="$(lsof -Pni UDP -p ${pid} -a | grep -m1 : | awk -F ":" '{print $2}')"
```

Comment:
This is not going to work reliably as you should not rely on port 53 being the first one reported by `lsof`.

Mind that FTL may bind to other port as well, such as 67 for DHCP or 547 for DHCPv6.
A quick check indeed suggests that DHCP ports will show up earlier (the order depends on the order `dnsmasq`/FTL bound to these ports) as both

https://github.com/pi-hole/FTL/blob/3b7bd2c5224c3ed13b983abcab26dd676cdc7cd0/src/dnsmasq/dnsmasq.c#L329

https://github.com/pi-hole/FTL/blob/3b7bd2c5224c3ed13b983abcab26dd676cdc7cd0/src/dnsmasq/dnsmasq.c#L348

are before

https://github.com/pi-hole/FTL/blob/3b7bd2c5224c3ed13b983abcab26dd676cdc7cd0/src/dnsmasq/dnsmasq.c#L422

I see three possibilities right now, alternatives are welcomed:

1. Simply ask FTL for its port over the Telnet API. This only needs a simple API extension in FTL.
2. Have FTL write the DNS port into some file which is then read by PHP. This is pretty indirect and the file could be outdated.
3. Check which port FTL bound to both on UDP and TCP. This *should* only happen for the DNS port. Even when this would work without an FTL change, it clearly looks like the most hacky solution among all three.

Reply:
Concerning my suggestion no.1: https://github.com/pi-hole/FTL/pull/1264

Reply:
Thanks for pointing out possible pitfalls. I think no. 1 is the most elegant one. I'll try to change the code accordingly.